packages/flutter/lib/src/rendering/proxy_sliver.dart

```diff
@@ -0,0 +1,366 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show Color;
+
+import 'package:flutter/foundation.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'object.dart';
+import 'sliver.dart';
+
+/// A base class for sliver render objects that resemble their children.
+///
+/// A proxy sliver has a single child and simply mimics all the properties of
+/// that child by calling through to the child for each function in the render
+/// sliver protocol. For example, a proxy sliver determines its geometry by
+/// asking its sliver child to layout with the same constraints and then
+/// matching the geometry.
+///
+/// A proxy sliver isn't useful on its own because you might as well just
+/// replace the proxy sliver with its child. However, RenderProxySliver is a
+/// useful base class for render objects that wish to mimic most, but not all,
+/// of the properties of their sliver child.
```

Comment:
Maybe add an see also to RenderProxyBox, since that seems to be doing the same in the box world? (Maybe also link back to this from RenderProxyBox)?